sickbeard/webserve.py

```diff
@@ -2693,20 +2694,47 @@ def massAddTable(self, rootDir=None):
                             if not indexer_id and i:
                                 indexer_id = i
 
-                cur_dir['existing_info'] = (indexer_id, show_name, indexer)
+                cur_dir['existing_info'] = {"indexer_id": indexer_id or '', "show_name": show_name or '',
+                                            "indexer": indexer or ''}
 
                 if indexer_id and Show.find(sickbeard.showList, indexer_id):
                     cur_dir['added_already'] = True
         return t.render(dirList=dir_list)
 
-    def newShow(self, show_to_add=None, other_shows=None, search_string=None):
+    def newShow(self, show_to_add=None, other_shows=None, search_string=None, **shows_to_add):
         """
```

Comment:
Needed to create some sort of hybrid solution for this. The **shows_to_add is added so it can accept the initial .ajax post with json data. After that, the form posting takes over. The form posting also uses json, but quoted_plus, for making sure it doesn't choke on the " or '. For like example Marvel's ....